numpy/core/src/umath/ufunc_object.c

```diff
@@ -3531,8 +3534,49 @@ reduce_loop(NpyIter *iter, char **dataptrs, npy_intp *strides,
         strides_copy[0] = strides[0];
         strides_copy[1] = strides[1];
         strides_copy[2] = strides[0];
-        innerloop(dataptrs_copy, countptr,
-                    strides_copy, innerloopdata);
+
+        if (!masked) {
+            innerloop(dataptrs_copy, countptr,
+                      strides_copy, innerloopdata);
+        }
+        else {
+            npy_intp count = *countptr;
+            char *maskptr = dataptrs[2];
+            char mask = *maskptr;
+            if (strides[2] != 0) {
```

Comment:
Can you add a `mask_stride` alias for this?

Reply:
Done